src/lib/OTA/OTA.h

```diff
@@ -17,20 +21,73 @@
 // Mask used to XOR the ModelId into the SYNC packet for ModelMatch
 #define MODELMATCH_MASK 0x3f
 
+#define SYNC_PACKET_SWITCH_OFFSET   1   // Switch encoding mode
+#define SYNC_PACKET_TLM_OFFSET      3   // Telemetry ratio
+#define SYNC_PACKET_RATE_OFFSET     6   // Rate index
+#define SYNC_PACKET_SWITCH_MASK     0b11
+#define SYNC_PACKET_TLM_MASK        0b111
+#define SYNC_PACKET_RATE_MASK       0b11
+
+typedef struct {
+    uint8_t type: 2,
+            crcHigh: 6;
+    union {
+        struct {
+            uint8_t fhssIndex;
+            uint8_t nonce;
+            uint8_t free:1,
+                    switchEncMode:2,
+                    newTlmRatio:3,
+                    rateIndex:2;
+            uint8_t UID3;
+            uint8_t UID4;
+            uint8_t UID5;
+        } sync;
+        struct {
+            uint8_t ch0High;
+            uint8_t ch1High;
+            uint8_t ch2High;
+            uint8_t ch3High;
+            uint8_t ch3Low:2,
+                    ch2Low:2,
+                    ch1Low:2,
+                    ch0Low:2;
+            uint8_t switches;
+        } rc;
+        struct {
+            uint8_t packageIndex;
+            uint8_t data[5];
+        } msp;
```

Comment:
We could use this refactoring step to come up with a better name for the msp package type since it does also transport other data types as well (from KISS and Ardupilot). I'm not 100% sure what would be a good name to describe the data but the existing names in crsf are telemetry frames and extended frames. A name like "data" sounds too generic

Reply:
Agreed! Fortunately there are space in this struct to add more types :)
There are many places which can be improved. I think that MSP improvement is an another PR...